### ISSUE TITLE
redis cache in cluster mode with version redis-py-cluster >= 2.0.0 not working

### ISSUE THREAD

**Issue by uwerler** [saltstack/salt#61842]
**Description**
When using redis cache in cluster mode setup the following trace shows up:

'''
[ERROR   ] An un-handled exception from the multiprocessing process 'Maintenance-8' was caught:                     
Traceback (most recent call last):                                                                                  
  File "/usr/lib/python3/dist-packages/salt/utils/process.py", line 1034, in wrapped_run_func                       
    return run_func()                                                                                               
  File "/usr/lib/python3/dist-packages/salt/master.py", line 225, in run                                            
    self.handle_presence(old_present)                                                                               
  File "/usr/lib/python3/dist-packages/salt/master.py", line 330, in handle_presence
    present = self.ckminions.connected_ids()       
  File "/usr/lib/python3/dist-packages/salt/utils/minions.py", line 633, in connected_ids
    search = self.cache.list("minions")            
  File "/usr/lib/python3/dist-packages/salt/cache/__init__.py", line 233, in list
    return self.modules[fun](bank, **self._kwargs)
  File "/usr/lib/python3/dist-packages/salt/loader/lazy.py", line 334, in __getitem__
    super().__getitem__(item)  # try to get the item from the dictionary
  File "/usr/lib/python3/dist-packages/salt/utils/lazy.py", line 105, in __getitem__
    raise KeyError(key)
KeyError: 'redis.list'
Process Maintenance-8:
Traceback (most recent call last):
  File "/usr/lib/python3.9/multiprocessing/process.py", line 315, in _bootstrap
    self.run()
  File "/usr/lib/python3/dist-packages/salt/utils/process.py", line 1034, in wrapped_run_func
    return run_func()
  File "/usr/lib/python3/dist-packages/salt/master.py", line 225, in run
    self.handle_presence(old_present)
  File "/usr/lib/python3/dist-packages/salt/master.py", line 330, in handle_presence
    present = self.ckminions.connected_ids()
  File "/usr/lib/python3/dist-packages/salt/utils/minions.py", line 633, in connected_ids
    search = self.cache.list("minions")
  File "/usr/lib/python3/dist-packages/salt/cache/__init__.py", line 233, in list
    return self.modules[fun](bank, **self._kwargs)
  File "/usr/lib/python3/dist-packages/salt/loader/lazy.py", line 334, in __getitem__
    super().__getitem__(item)  # try to get the item from the dictionary
  File "/usr/lib/python3/dist-packages/salt/utils/lazy.py", line 105, in __getitem__
    raise KeyError(key)
KeyError: 'redis.list'
'''

**Setup**
cache.redis.cluster_mode: true
cache.redis.cluster.skip_full_coverage_check: true
cache.redis.cluster.startup_nodes:
  - host: cluster-node-1
    port: 6379
  - host: cluster-node-2
    port: 6379
  - host: cluster-node-3
    port: 6379
cache.redis.db: '0'

Please be as specific as possible and give set-up details.

- [ ] on-prem machine
- [x] VM (VMWare)
- [ ] VM running on a cloud service, please be explicit and add details
- [ ] container (Kubernetes, Docker, containerd, etc. please specify)
- [ ] or a combination, please be explicit
- [ ] jails if it is FreeBSD

  
**Steps to Reproduce the behavior**
Configure redis cache in cluster mode with module version redis-py-cluster >= 2.0.0

**Expected behavior**
Getting minion data stored in redis.

**Screenshots**

**Versions Report**
<details><summary>salt --versions-report</summary>
(Provided by running salt --versions-report. Please also mention any differences in master/minion versions.) 

```yaml
root@lxsalt03:~# salt --versions-report
Salt Version:
           Salt: 3004
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: unknown
       dateutil: 2.8.1
      docker-py: Not Installed
          gitdb: 4.0.5
      gitpython: 3.1.14
         Jinja2: 2.11.3
        libgit2: 1.1.0
       M2Crypto: Not Installed
           Mako: Not Installed
        msgpack: 1.0.0
   msgpack-pure: Not Installed
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: Not Installed
   pycryptodome: 3.9.7
         pygit2: 1.4.0
         Python: 3.9.2 (default, Feb 28 2021, 17:03:44)
   python-gnupg: Not Installed
         PyYAML: 5.3.1
          PyZMQ: 20.0.0
          smmap: 4.0.0
        timelib: Not Installed
        Tornado: 4.5.3
            ZMQ: 4.3.4
 
Salt Extensions:
 saltext.vmware: 22.2.2.0rc1
 
System Versions:
           dist: debian 11 bullseye
         locale: utf-8
        machine: x86_64
        release: 5.10.0-8-amd64
         system: Linux
        version: Debian GNU/Linux 11 bullseye
```
pip list:

```yaml
Package           Version
----------------- --------------
apache-libcloud   3.2.0
cached-property   1.5.2
certifi           2020.6.20
chardet           4.0.0
CherryPy          8.9.1 
contextvars       2.4
crc16             0.1.1
croniter          0.3.34
cryptography      3.3.2
distro            1.5.0
dnspython         2.0.0
gitdb             4.0.5
GitPython         3.1.14
gpg               1.14.0-unknown
httplib2          0.18.1
idna              2.10
immutables        0.14
Jinja2            2.11.3
lockfile          0.12.2
Markdown          3.3.4
MarkupSafe        1.1.1
msgpack           1.0.0
natsort           7.1.0
netaddr           0.7.19
pip               20.3.4
psutil            5.8.0
psycopg2          2.8.6
pycryptodomex     3.9.7
pycurl            7.43.0.6
pygit2            1.4.0
Pygments          2.7.1
pyinotify         0.9.6
pyOpenSSL         20.0.1
PySimpleSOAP      1.16.2
python-apt        2.2.1
python-consul     0.7.1
python-dateutil   2.8.1
python-debian     0.1.39
python-debianbts  3.1.0
python-redis      0.3.0
pytz              2021.1
pyvmomi           7.0.2
PyYAML            5.3.1
pyzmq             20.0.0
raven             6.10.0
redis             3.5.3
redis-py-cluster  2.1.3
reportbug         7.10.3+deb11u1
repoze.lru        0.7
requests          2.25.1
requests-toolbelt 0.9.1
Routes            2.5.1
salt              3004
saltext.vmware    22.2.2.0rc1
setuptools        52.0.0
simplejson        3.17.2
six               1.16.0
smmap             4.0.0
systemd-python    234
tzlocal           2.1
urllib3           1.26.5
WebOb             1.8.6
wheel             0.34.2
```
</details>

**Additional context**

**Comment by uwerler** [comment]
Fix provided by PR #61843.

**Comment by dmurphy18** [comment]
This issue is waiting on PR https://github.com/saltstack/salt/pull/62841 which still needs tests before being accepted.

**Comment by dmurphy18** [comment]
@uwerler Closing this issue due to lack of response, please re-open and provide the request tests if you desire this and the associated #62481 to be considered, also closed associated PR #62481
